modules/core/src/command_line_parser.cpp

```diff
@@ -134,22 +135,25 @@ void CommandLineParser::getByIndex(int index, bool space_delete, int type, void*
                 String v = impl->data[i].def_value;
                 if (space_delete == true) v = impl->cat_string(v);
 
-                // it is an error if we just got the default value here
-                if(v.empty())
-                    break;
-
+                // the key was neither specified nor has it a default value
+                if(v.empty()) {
```

Comment:
Please use here this condition: `if (v.empty() && type != Param::STRING)`
Default empty string is fine and not an error. Conversion problem is not raised.

Problem is here: http://pullrequest.opencv.org/buildbot/builders/precommit_linux64/builds/4332/steps/perf_core/logs/stdio